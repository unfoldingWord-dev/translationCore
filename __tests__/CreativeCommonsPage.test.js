/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CreativeCommonsPage from '../src/js/components/home/usersManagement/pages/CreativeCommonsPage';

beforeAll(() => {
  configure({ adapter: new Adapter() });
});

test('snapshot', () => {
  const wrapper = renderer.create(
    <CreativeCommonsPage translate={key => key} onBackClick={jest.fn()}/>
  );
  expect(wrapper).toMatchSnapshot();
});

test('back', () => {
  const backCallback = jest.fn();
  const wrapper = shallow(
    <CreativeCommonsPage translate={key => key} onBackClick={backCallback}/>
  );
  wrapper.find('button').simulate('click');
  expect(backCallback).toBeCalled();
});
