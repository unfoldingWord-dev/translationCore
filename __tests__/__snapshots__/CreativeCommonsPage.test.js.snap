// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <button
    className="btn-second"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-share-alt"
      style={
        Object {
          "transform": "scaleX(-1)",
        }
      }
    />
     
    buttons.back_button
  </button>
  <div
    style={
      Object {
        "alignItems": "center",
        "color": "black",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <h4>
      <b>
        Creative Commons Attribution-ShareAlike 4.0 License
      </b>
    </h4>
    <p
      style={
        Object {
          "padding": "15px",
        }
      }
    >
      This is a human-readable summary of (and not a substitute for) the 
      <a
        href="https://creativecommons.org/licenses/by-sa/4.0/legalcode"
        style={
          Object {
            "color": "black",
            "cursor": "pointer",
            "textDecoration": "underline",
          }
        }
      >
        license
      </a>
    </p>
    <h4>
      <b>
        You are free to:
      </b>
    </h4>
    <p>
      <b>
        Share
      </b>
       — copy and redistribute the material in any medium or format.
    </p>
    <p>
      <b>
        Adapt
      </b>
       — remix, transform, and build upon the material  for any purpose, even commercially. This license is acceptable for Free Cultural Works.
    </p>
    <p>
      <i>
        The licensor cannot revoke these freedoms as long as you follow the license terms.
      </i>
    </p>
    <h4>
      <b>
        Under the following terms:
      </b>
    </h4>
    <p>
      <b>
        Attribution
      </b>
       — You must attribute the work as follows: "Original work available  at https://git.door43.org/." Attribution statements in derivative works should not in any way  suggest that we endorse you or your use of this work
    </p>
    <p>
      <b>
        ShareAlike
      </b>
       — If you remix, transform, or build upon the material, you must  distribute your contributions under the same license as the original.
    </p>
    <p>
      <b>
        No additional restrictions
      </b>
       — You may not apply legal terms or technological  measures that legally restrict others from doing anything the license permits.
    </p>
    <h4>
      <b>
        Notices:
      </b>
    </h4>
    <p>
      You do not have to comply with the license for elements of the material in the  public domain or where your use is permitted by an applicable exception or limitation.
    </p>
    <p>
      No warranties are given. The license may not give you all of the permissions  necessary for your intended use. For example, other rights such as publicity,  privacy, or moral rights may limit how you use the material.
    </p>
  </div>
</div>
`;
